src/scripts/list-users.test.ts

```diff
@@ -0,0 +1,176 @@
+/* eslint-disable functional/no-return-void */
+/* eslint-disable functional/no-expression-statement */
+/* eslint-disable functional/functional-parameters */
+import * as TE from 'fp-ts/TaskEither';
+import * as O from 'fp-ts/Option';
+import { UserService } from './services/user-service';
+import { users, usersInGroup } from './list-users';
+import { GroupService } from './services/group-service';
+
+import * as test from './__fixtures__/data-providers';
+
+describe('Listing all users', () => {
+  // Note that listing multiple or no users is outside the scope of the test, as that functionality comes from a dependency
+  it('passes when attempting to list users without a group without error', async () => {
+    // Given a user service that can retreive a list containing a single user
+    const userService: UserService = {
+      ...test.baseUserService(),
+      listUsers: () => TE.right([test.user]),
+    };
+
+    // When we attempt to list the users from the client
+    const result = await users(userService)();
+
+    // Then we should have a right
+    expect(result).toBeRight();
+
+    // And if it is a right, we should get a string as a result
+    const actual = result._tag === 'Right' ? result.right : result.left;
+
+    // And we expect that the resulting string contains the ID, Email, Name, and Status of the User
+    expect(actual).toContain('user_id');
+    expect(actual).toContain('test@localhost');
+    expect(actual).toContain('test user');
+    expect(actual).toContain('ACTIVE');
+  });
+
+  it('fails when attempting to list users without a group with error', async () => {
+    // Given a user service that fails to retrieve a list of users
+    const userService: UserService = {
+      ...test.baseUserService(),
+      listUsers: () => TE.left('No users.'),
```

Comment:
Change `No users.` to `expected error` here and else where.

Reply:
Modified